osu.Game.Tests/Visual/Navigation/TestSceneScreenNavigation.cs

```diff
@@ -575,6 +575,19 @@ public void TestExitGameFromSongSelect()
             AddStep("test dispose doesn't crash", () => Game.Dispose());
         }
 
+        [Test]
+        public void TestExitIsBlocked()
```

Comment:
The fix itself appears to be legitimate and correct upon manual testing, but I'm not sure whether this test is covering the failure case correctly, given that it passes for me at b6beb0e9bd3a5cb5fbdf56a1f2be65f1613a6b3d already (i.e. without the fix in place).